src/test/java/io/apicurio/datamodels/LibraryTest.java

```diff
@@ -62,4 +74,59 @@ public void testReadDocumentFromJSONString() {
         Assert.assertEquals("3.0.2", ((Oas30Document) doc).openapi);
     }
 
+    @Test
+    public void testValidate() {
+        String jsonString = "{\r\n" +
+                "    \"openapi\": \"3.0.9\",\r\n" +
+                "    \"info\": {\r\n" +
+                "        \"title\": \"Very Simple API\",\r\n" +
+                "        \"version\": \"1.0.0\"\r\n" +
+                "    }\r\n" +
+                "}";
+        Document doc = Library.readDocumentFromJSONString(jsonString);
+
+        List<ValidationProblem> problems = Library.validate(doc, null);
+        Assert.assertTrue(problems.size() > 0);
+    }
+
+    private final CountDownLatch lock = new CountDownLatch(1);
+
+    @Test
+    public void testValidateWithExtensions() {
+        String jsonString = "{\r\n" +
+                "    \"openapi\": \"3.0.2\",\r\n" +
+                "    \"info\": {\r\n" +
+                "        \"title\": \"Very Simple API\",\r\n" +
+                "        \"version\": \"1.0.0\"\r\n" +
+                "    }\r\n" +
+                "}";
+        Document doc = Library.readDocumentFromJSONString(jsonString);
+
+        List<IValidationExtension> extensions = new ArrayList<>();
+        extensions.add(new TestValidationExtension());
+
+        CompletableFuture<List<ValidationProblem>> problems = Library.validateWithExtensions(doc, null, extensions);
+        List<ValidationProblem> receivedProblems = new ArrayList<>();
+        problems.whenCompleteAsync((pList, done) -> {
+            receivedProblems.addAll(new ArrayList<>(pList));
+            lock.countDown();
+        });
+        Optional<ValidationProblem> extensionProblem = receivedProblems.stream().
+                filter(p -> Objects.equals(p.errorCode, "TEST-001")).
+                findFirst();
+        Assert.assertTrue(extensionProblem.isPresent());
```

Comment:
@EricWittmann good news - I've managed to get a PromiseAdapter working so that CompletableFuture transpiles to a JavaScript Promise!

In the process I seem to have broken the JavaScript implementation though, need to look into that and cover it with a test case.

Reply:
Cool progress!!

Reply:
The problem is likely the upgrade to the JSweet version.  See my comment above and the link to a JSweet GH issue I opened.

Reply:
Yeah, i have since resolved that, I think it was down to the version number.